research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,233 @@
+
+### Mouse interaction
+
+> Define the mouse behaviors of the component.
+>
+> - What specific mouse events (down, up, move) are being listened to and how does that affect the component?
+> - For behaviors which are accessible through mouse input, can you also do the same thing using a keyboard?
+>
+> Example:
+>
+> - `mousedown` should immediately attempt to set value to the appropriate value.
+> - `mousemove` should be attached to the window during mousedown to track window-wise move events. Value should be updated appropriately. Note; if a `mousemove` occurs
+>   without the primary button pressed, tracking should cancel and treat the even as a `mouseup`. (Edge case: mouse down, move cursor out of window, release mouse)
+> - `mouseup` should remove the `mousemove` event.
+
+### Touch interaction
+
+> Define the touch behaviors of the component.
+>
+> - Are there different behaviors than in mouse scenarios?
+> - Are touch targets large enough for fat fingers? (44x44px is the recommended minimum size.) For example, the `thumb` of a slider might appear to be small, but the touchable area should be larger.
+
+### Screen reader accessibility
+
+> Review/link aria specifications and list which aria labels should appear on which parts of the component.
+>
+> If there are ambiguous requirements or lacking guidance in the specifications, consult the experts in your organization to propose refinements/additions to the specifications.
+>
+> Example:
+>
+> `root`:
+>
+> - should render the native element using the `as` prop, defaulting to `div`
+> - should mix in native props expected for the element type defined in `as`.
+>
+> `thumb`:
+>
+> - should be focusable via `tabindex=0`
+> - role set to `{ComponentName}`
+> - receives `aria-valuemin` and `aria-valuemax` representing min/max
+> - receives `aria-valuenow` representing the current value
+
+### Accessibility concerns for the user
+
+> Consider which props, mapped to attribute values, must be provided for the user to make the component accessible. Often props which require localization or calculating a value must be provided by the user.
+>
+> For React props, consider using `ariaLabel` and `ariaLabelledBy`, rather than the natural kebab-cased variants (`aria-label`) so that they can be provided without destroying type safety.
+
+## Themability and customization
+
+> Components should be built to withstand multiple aesthetic looks. This ensures customizability and reuse in the long term.
+>
+> At minimum, each component should consider the following theming requirements: light mode, dark mode, high contrast light and high contrast dark.
+>
+> If applicable, include some screenshots of themed variants of the component to get an idea of what is to be achieved through customization.
+
+### Class names
+
+> Components should include a class name contract to achieve maximum reusability. The contract should include:
```

Comment:
Is there a correlation here to the DOM structure as well? Not sure how we tie the two together but these seem linked if the goal is reusability.

Reply:
DOM consists of elements which would be 1 to 1 with slot names. A slot would be 1 to 1 with class name, so yes there is a correlation. One slot = one class.

Additionally for stateful things, we would keep state based classes attached to the root of a component, so that selectors can be used to adjust child content based on component state.

Does that seem reasonable?

Reply:
Class names may be too much about implementation detail in terms of the styling system.

Reply:
For sure get it here; I'm open, more asking the question. I'm not sure styles/classNames are reusable if the structure isn't 1:1, but that's also more of something we've run into specific to our implementation. I'm good either way.